climpred/tests/test_compute_dims.py

```diff
@@ -0,0 +1,196 @@
+import pytest
+from climpred.bootstrap import bootstrap_hindcast, bootstrap_perfect_model
+from climpred.constants import (
+    PM_COMPARISONS,
+    PROBABILISTIC_HINDCAST_COMPARISONS,
+    PROBABILISTIC_PM_COMPARISONS,
+    PROBABILISTIC_METRICS,
+)
+from climpred.prediction import compute_hindcast, compute_perfect_model
+from climpred.tutorial import load_dataset
+from climpred.utils import get_comparison_function
+from xarray.testing import assert_allclose
+
+# Test apply_metric_to_member_dim
+
+
+@pytest.fixture
+def pm_da_ds1d():
+    da = load_dataset('MPI-PM-DP-1D')
+    da = da['tos'].isel(area=1, period=-1)
+    return da
+
+
+@pytest.fixture
+def pm_da_control1d():
+    da = load_dataset('MPI-control-1D')
+    da = da['tos'].isel(area=1, period=-1)
+    return da
+
+
+@pytest.fixture
+def initialized_da():
+    da = load_dataset('CESM-DP-SST')['SST']
+    da = da - da.mean('init')
+    return da
+
+
+@pytest.fixture
+def uninitialized_da():
+    da = load_dataset('CESM-LE')['SST']
+    # add member coordinate
+    da['member'] = range(1, 1 + da.member.size)
+    da = da - da.mean('time')
+    return da
+
+
+@pytest.fixture
+def observations_da():
+    da = load_dataset('ERSST')['SST']
+    da = da - da.mean('time')
+    return da
+
+
+@pytest.mark.parametrize('stack_dims', [True, False])
+@pytest.mark.parametrize('comparison', PROBABILISTIC_PM_COMPARISONS)
+def test_pm_comparison_stack_dims(pm_da_ds1d, comparison, stack_dims):
+    comparison = get_comparison_function(comparison, PM_COMPARISONS)
+    actual_f, actual_r = comparison(pm_da_ds1d, stack_dims=stack_dims)
```

Comment:
Note that in Codacy, it mentions `actual_r` isn't used here. Is it supposed to be used or no? If not you could just make it an underscore return .. `actual_f, _ = comparison(...)`. Doesn't really matter though. Just pointing it out.